### PR TITLE
feat: [PL-41444]: Change mark for deletion day 90 -> 60

### PR DESCRIPTION
**Pull request by sahibaHarness**
## Describe your changes
Presently, as per our contracts Harness needs to delete Account data within 90 days following churn or expiration.
Currently, we modify the account status to “MARKED-FOR-DELETION” after a 90-day period. Consequently, data deletion starts on next weekend if the account is picked by the iterator. But at times due to unforeseen issues this can get delayed by more than a week or two.
Hence to make sure that we are abiding by the contract; we are going to change the period of marking for deletion from 90 days to 60 days. This will give us good window to clear the data from our end
image
image
image
image
image

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/54085)

### CODE REVIEW

**Comment by sahibaHarness** [comment]
trigger uts

**Review by pbarapatre10 — APPROVED** [review]
Please test the `Harness trial has expired` email(s) going out post this change (will fewer emails be sent? to trial account user) and get this properly updated in docs, and communicated to CSMs.

**Comment by sahibaHarness** [comment]
trigger uts

**Comment by sahibaHarness** [comment]
trigger ut3,ut7

**Comment by sahibaHarness** [comment]
trigger uts

**Comment by sahibaHarness** [comment]
trigger ut2

**Review by ankit-harness — APPROVED** [review]
:lgtm:

> Reviewed 9 of 9 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/54085#-:-NhAbOk75nUrsehGzXdZ:bnfp4nl)* status: 10 file(s) to review. 1 of 2 approvals obtained.

File: 400-rest/src/test/java/software/wings/licensing/LicenseServiceTest.java [hunk 1]
```diff
@@ -460,18 +460,18 @@ public void shouldCheckForLicenseExpiry() throws InterruptedException {
...
   @Category(UnitTests.class)
   public void getEmailTemplateNameForTrialAccountExpiration() {
     long currentTime = System.currentTimeMillis();
     long expiryTime = currentTime + 1000;
     long oneDayAfterExpiry = expiryTime + oneDayTimeDiff;
     long twoDaysAfterExpiry = expiryTime + (2 * oneDayTimeDiff);
     long thirtyDaysAfterExpiry = expiryTime + (30 * oneDayTimeDiff);
+    long fortyFiveDaysAfterExpiry = expiryTime + (45 * oneDayTimeDiff);
+    long fiftyNineDaysAfterExpiry = expiryTime + (59 * oneDayTimeDiff);
     long sixtyDaysAfterExpiry = expiryTime + (60 * oneDayTimeDiff);
```
Comment (pbarapatre10) [inline 400-rest/src/test/java/software/wings/licensing/LicenseServiceTest.java:473]: nit: Do we need both 59 & 60? or should just one will be sufficient, what is the difference in mail/subject in both the cases

Comment (pbarapatre10) [reply 400-rest/src/test/java/software/wings/licensing/LicenseServiceTest.java:473]: This is a question BTW

**Review by pbarapatre10 — APPROVED** [review]

File: 400-rest/src/test/java/software/wings/licensing/LicenseServiceTest.java (hunk 1 above)
Comment (sahibaHarness) [reply 400-rest/src/test/java/software/wings/licensing/LicenseServiceTest.java:473]: We need both to adhere with the older flow. Had a discussion with @batraprashant; we are not changing any thing a part from the dates.

**Pull request merged**